utils/spdx/src/main/kotlin/model/SpdxSnippet.kt

```diff
@@ -82,11 +82,38 @@ data class SpdxSnippet(
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
     val name: String = "",
 
+    /**
+     * The ranges in the original host file that the snippet information applies to.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val ranges: List<Range> = emptyList(),
+
     /**
      * The SPDX reference referencing the document within the SpdxDocument containing the snippet.
      */
     val snippetFromFile: String
 ) {
+    data class Range(
+        /** The start of the snippet range. */
+        val startPointer: Pointer,
+
+        /** The end of the snippet range. */
+        val endPointer: Pointer
+    )
```

Comment:
I couldn't find 100% proof in the spec, but I believe the intention of it is that the range is 
either 1. a line range or 2.  a byte range. This implies that `startPointer` and `endPointer`
must be of the same type (byte range vs line range).

Looking at the `Pointer` below, it should hold `lineNumber==null xor offset==null`. So, the model could be stronger typed also in terms of not null guarantee, if it modeled `ByteRangerPointer` and `LineRangePointer` separately.

Do you think it would make sense to add polymorphism here? (I'm not sure myself if this is a good approach, an alternative could be to add more documentation and also invariant checks into the cosntructor)

Reply:
I was playing with having `sealed class Pointer` with derived `LineNumberPointer` and `ByteOffsetPointer`, but as either would then only have one `Int?` property, we'd run into a deserializer exception if someone specified both types of pointers, instead of having a probably nicer message triggered by `require` checks. I'm not sure what's better here.

This is BTW again an unfortunate example of SPDX model design choices, ping @goneall.

Reply:
See https://github.com/oss-review-toolkit/ort/pull/7890 for the split-out SPDX 2.2 changes.